src/helpers/UserProvider.js

```diff
@@ -0,0 +1,46 @@
+import { createContext, useEffect, useState } from "react";
+import axios from "axios";
+import { useNavigate } from "react-router-dom";
+
+
+export const UserContext = createContext()
+
+const UserProvider = ({children}) => {
+  
+  const [user,setUser] = useState();
+  const [loggedIn,setLoggedIn] = useState(localStorage.getItem("X-AccessToken"))
+
+  let navigate = useNavigate();
+
+  useEffect(() => {
```

Comment:
@DEONSKY  Burası ne için kullanılıyor?

Reply:
Token durumuna göre yönlendirme yapmakta